ShopAdmin/Commands/Category.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.Extensions.Logging;
+using ShopGeneral.Services;
+
+namespace ShopAdmin.Commands
+{
+    public class Category : ConsoleAppBase
+    {
+        private readonly ILogger<Category> _logger;
+        private readonly ICategoryService _categoryService;
+        private readonly IProductService _productService;
+
+        public Category(ILogger<Category> logger, ICategoryService categoryService, IProductService productService)
+        {
+            _logger = logger;
+            _categoryService = categoryService;
+            _productService = productService;
+        }
+
+        public void Checkempty()
+        {
+            _logger.LogInformation("CheckEmpty starting");
+
+            string directoryPath = "outfiles\\category\\";
+            string filePath = $"{directoryPath}missingproducts-{DateTime.UtcNow:yyyyMMdd}.txt";
+
+            var categories = _categoryService.GetTrendingCategories(100); 
+            var products = _productService.GetAllProducts(); 
+                                           
+            List<string> categoriesWithNoProducts = new();
+
+            foreach (var category in categories)
+            {
+                bool isFound = false;
+                foreach (var product in products)
+                {
+                    if (product.CategoryId == category.Id)
+                    {
+                        Console.WriteLine($"Category {category.Id} {category.Name} has at least one product linked.");
+                        isFound = true;
+                        break;
+                    }
+                }
+
+                if(!isFound)
+                {
+                    Console.WriteLine($"Category {category.Id} {category.Name} has at least one product linked.");
```

Comment:
Does not reflect what's happening. Change to i.e. "does not have a product linked..."